### DISCUSSION TITLE
How to decipher fields in database

### DISCUSSION THREAD

**Discussion by wiljateiwaz** [dani-garcia/vaultwarden#3364] [Q&A] [answered]
Hi,

I'm migrating my Keepass db to Vaultwarden and when I did the import of everything, few metadata were not kept. The most important two for me are the **_creation date_** and the **_history_** of an entry. I understand the history might be harder to handle, so I'm focusing first on the **_creation date_**.

So I exported the `db.sqlite3` and opened it with SQLite DB Browser, I can see the list of my `ciphers` with the metadata I'm looking for, now I'm looking for a way to decipher the content of my table with my password to be able to find the right entry to modify.

Is there a way to manage that?

_Edit: I only need to decipher the column **name**._

**Comment by BlackDex** [comment]
Column name? Those aren't encrypted. The database it self, in regards to the schema is just plain/text.

If you want to encrypt/decrypt entries, you are better of to just login and just any client ;).

When you really want to do it your self,  you need to read the following page: https://bitwarden.com/help/what-encryption-is-used/
And more specifically https://bitwarden.com/crypto.html

**Comment by wiljateiwaz** [comment, marked as answer]
I meant the **name** of the **entries** in the table `ciphers`.

> If you want to encrypt/decrypt entries, you are better of to just login and just any client ;).

How is that supposed to help me changed the appropriate creation time of a specific entry?

I checked the links, but they're not really precise about what's done. I tried a tool to decipher AES-256 CBC already, with the entry name enciphered as an input, and the symmetric key. Or even reverse the process, trying to reproduce the enciphered data. Both unsuccessful, hence my request here. 😉

**Reply by kpfleming** [reply]
All of that encryption is done by the Bitwarden client, not the Vaultwarden server. If you want to reverse that, you'll need to learn how the client did it.

**Reply by wiljateiwaz** [reply]
I figured that's the way, and I even checked their [security whitepaper](https://bitwarden.com/images/resources/security-white-paper-download.pdf), but I thought to ask here for help to _maybe_ solve it faster. Also I can't really ask Bitwarden's support since I'm never suppose to have access to the DB there.

**Reply by BlackDex** [reply]
I'm afraid that what you want is not very easy.
If you go to a specific cipher in the web-vault in the URL there should be the uuid of that specific cipher. But that would be a lot of manual work.

The only way i can think of is the following by using the Bitwarden CLI.
```bash
bw list items | jq -rM '.[] | [.name, .id]|@tsv'
```

That will output all your passwords `names` with there corresponding `uuid`, which you can then match in the database.

**Reply by wiljateiwaz** [reply]
It's pretty helpful, out of this I started to make a script to match names from this list with the keepass, extract the creation date from the DB _via_ `uuid` and compare it so I know which I should change. It's gonna be already pretty good this way.

Thanks for this!
